neonix/io/nixio.py

```diff
@@ -0,0 +1,270 @@
+                nix_block = self.nix_file.blocks[0]
+                return self.add_recordingchannelgroup(rcg, nix_block)
+                break
+        else:
+            raise LookupError(
+                    "Parent Block with name '{}' for RecordingChannelGroup "
+                    "with name '{}' does not exist in file '{}'.".format(
+                            parent_block.name, rcg.name, self.filename))
+
+    def add_segment(self, segment, parent_block):
+        """
+        Write the provided ``segment`` to the NIX file as a child of
+        parent_block after converting to a ``Group`` object.
+
+        :param segment: Neo segment to be written
+        :param parent_block: The parent NIX Block
+        :return: The newly created NIX Group
+        """
+        nix_name = segment.name
+        nix_type = "neo.segment"
+        nix_definition = segment.description
+        nix_group = parent_block.create_group(nix_name, nix_type)
+        nix_group.definition = nix_definition
+        if segment.rec_datetime:
+            # Truncating timestamp to seconds
+            nix_group.force_created_at(int(segment.rec_datetime.timestamp()))
+        if segment.file_datetime:
+            group_metadata = self._get_or_init_metadata(nix_group)
+            # Truncating timestamp to seconds
+            group_metadata .create_property(
+                    "neo.file_datetime",
+                    nix.Value(int(segment.file_datetime.timestamp())))
+        if segment.file_origin:
+            group_metadata = self._get_or_init_metadata(nix_group)
+            group_metadata.create_property("neo.file_origin",
+                                           nix.Value(segment.file_origin))
+        return nix_group
+
+    def add_recordingchannelgroup(self, rcg, parent_block):
+        """
+        Write the provided ``rcg`` (RecordingChannelGroup) to the NIX file as
+        a child of ``parent_block`` after converting to a ``Source`` object.
+
+        :param rcg: The Neo rcg to be written
+        :param parent_block: The parent neo block of the provided rcg
+        :return: The newly created NIX Source.
+        """
+        nix_name = rcg.name
+        nix_type = "neo.recordingchannelgroup"
+        nix_definition = rcg.description
+        nix_source = parent_block.create_source(nix_name, nix_type)
+        nix_source.definition = nix_definition
+        if rcg.file_origin:
+            source_metadata = self._get_or_init_metadata(nix_source)
+            source_metadata.create_property("neo.file_origin",
+                                            nix.Value(rcg.file_origin))
+        if hasattr(rcg, "coordinates"):
+            source_metadata = self._get_or_init_metadata(nix_source)
+            coordinates = rcg.coordinates  # should this be copied?
+            source_metadata.create_property("neo.coordinates",
+                                            nix.Value(coordinates))
```

Comment:
I'm sure it doesn't. I thought I'd skip spending time on anything involving the RCG and its attributes until everything about it is clear.
As for the coordinates, a placeholder method, with an issue to implement conversion/copying is probably best to act as a reminder.